### PR TITLE
 #557761: fixed sitemap handler

### PR DESCRIPTION
**Pull request by matkovskyi**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by StanislavStativa — APPROVED** [review]

File: packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/api/sitemap.ts [hunk 1]
```diff
@@ -19,24 +19,40 @@ const sitemapApi = async (
...
+          responseType: 'stream',
+        })
+        .then((response: AxiosResponse) => {
+          response.data.pipe(res);
+        })
+        .catch(() => res.redirect('/404'));
+    }
+  } else {
+    const sitemaps = await sitemapXmlService.fetchSitemaps();
+    const SitemapLinks = sitemaps
```
Comment (ambrauer) [inline packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/api/sitemap.ts:41]: Do we need to handle case where `sitemaps` is empty?

**Review by ambrauer — CHANGES_REQUESTED** [review]
See couple comments. In general, would also be good to see some comments to help a developer understand the logic in here. There is more going on now... See base nextjs initializers' [api routes](https://github.com/Sitecore/jss/tree/687d9883da9484d66c5f8fb1bf92ed9c66cf0c6f/packages/create-sitecore-jss/src/templates/nextjs/src/pages/api) for examples.

File: packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/api/sitemap.ts (hunk 1 above)
Comment (matkovskyi) [reply packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/api/sitemap.ts:41]: When we try to open sitemap.xml link without sitemap-1.xml

Comment (matkovskyi) [reply packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/api/sitemap.ts]: I discussed with Alan and we decided if sitemaps is empty the script will redirect to 404 page

**Review by ambrauer — APPROVED** [review]
Looks good

**Pull request merged**